res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -119,6 +193,35 @@ Prime4.Deck = function(deckNumbers, midiChannel) {
 
 Prime4.Deck.prototype = new components.Deck();
 
+// Headphone Cue / PFL buttons for Decks 1 and 2
+
+// eslint-disable-next-line-no-var
+var headphoneCue = [];
+
+// eslint-disable-next-line-no-var
+for (var i = 1; i <= 4; i++) {
+    headphoneCue[i] = new components.Button({
+        midi: [0x8F + i, 0x0D],
+        group: "[Channel" + i + "]",
+        key: "pfl",
+        on: eval("colDeck" + i),
+        off: eval("colDeck" + i + "Dark"),
+        type: components.Button.prototype.types.toggle,
+    });
+}
```

Comment:
Instead of instantiating these a bunch of at the top level, put one pfl button into the deck, which then gets instantiated a bunch of times.

Reply:
The `headphoneCue` elements are part of the Prime 4's centre mixer, whereas there are only 2 physical decks on the Prime 4 that get toggled between with buttons. If I add a `headphoneCue` button into the `Prime4.Deck` container, I can only activate the pfl that corresponds to the deck that's currently active. In other words, I can't use Deck 3's PFL button if Deck 1 is active on the left side of the unit, or Deck 4's PFL button if Deck 2 is active on the right side. According to the documentation's examples I'm doing everything correctly AFAIK, so how would I go about implementing this? A lot of other mapping examples have used 4 different `Deck` containers, but the Prime 4 doesn't change its outgoing MIDI values even if you press the deck toggle or shift buttons, unlike almost every other controller I have seen mappings for inside Mixxx. What should I do?

Reply:
Ah, I understand. The issue here is now that we only have 2 physical decks but 4 physical mixer strips. So the solution is to create a mixer strip component container that contains all the things in the mixer (EQ, FX switches, Filter, PFL, volume) and instantiate 4 copies of those *outside of the deck*. So then calling `Deck.toggle()` will not change the group of the controls in there.

Reply:
OK cool, that makes sense. I'll give this a shot when I'm next available. Thanks!